src/west/configuration.py

```diff
@@ -660,45 +703,50 @@ def _location(cfg: ConfigFile, topdir: PathType | None = None, find_local: bool
             # See https://github.com/zephyrproject-rtos/west/issues/300
             # for details.
             pd = PureWindowsPath(os.environ['ProgramData'])
-            return os.fspath(pd / 'west' / 'config')
+            return [os.fspath(pd / 'west' / 'config')]
 
         raise ValueError('unsupported platform ' + plat)
     elif cfg == ConfigFile.GLOBAL:
         if 'WEST_CONFIG_GLOBAL' in env:
-            return _rel_topdir_to_abs(env['WEST_CONFIG_GLOBAL'], topdir)
+            paths = _Converter.str_to_paths(env['WEST_CONFIG_GLOBAL'])
+            return [_rel_topdir_to_abs(p, topdir) for p in paths]
 
         if platform.system() == 'Linux' and 'XDG_CONFIG_HOME' in env:
-            return os.path.join(env['XDG_CONFIG_HOME'], 'west', 'config')
+            return [os.path.join(env['XDG_CONFIG_HOME'], 'west', 'config')]
 
-        return os.fspath(Path.home() / '.westconfig')
+        return [os.fspath(Path.home() / '.westconfig')]
     elif cfg == ConfigFile.LOCAL:
         if 'WEST_CONFIG_LOCAL' in env:
-            return _rel_topdir_to_abs(env['WEST_CONFIG_LOCAL'], topdir)
+            paths = _Converter.str_to_paths(env['WEST_CONFIG_LOCAL'])
+            return [_rel_topdir_to_abs(p, topdir) for p in paths]
 
         if topdir:
-            return os.fspath(Path(topdir) / WEST_DIR / 'config')
+            return [os.fspath(Path(topdir) / WEST_DIR / 'config')]
 
         if find_local:
             # Might raise WestNotFound!
-            return os.fspath(Path(west_dir()) / 'config')
+            return [os.fspath(Path(west_dir()) / 'config')]
         else:
-            return ''
+            return []
     else:
         raise ValueError(f'invalid configuration file {cfg}')
 
 
-def _gather_configs(cfg: ConfigFile, topdir: PathType | None) -> list[str]:
+def _gather_configs(cfg: ConfigFile, topdir: PathType | None) -> list[Path]:
```

Comment:
Not sure if we need to change the return type here? You can simply replace `append` with `extend` in this function and be done with it.